test/units/TEST-65-ANALYZE.sh

```diff
@@ -1105,6 +1105,40 @@ else
     echo "have no tpm2"
 fi
 
+# check systemd-analyze unit-shell with a namespaced unit
+echo "Checking systemd-analyze unit-shell with PrivateTmp"
+UNIT_NAME="test-unit-shell.service"
+UNIT_FILE="/etc/systemd/system/$UNIT_NAME"
+cat >"$UNIT_FILE" <<EOF
+[UNIT]
+Description=Test unit for systemd-analyze unit-shell
+[SERVICE]
+ExecStart=/bin/sh -c "echo 'Hello from test unit' > /tmp/testfile; sleep infinity"
+PrivateTmp=yes
+EOF
+# Start the service
+systemctl start "$UNIT_NAME"
+# Wait for the service to be active
+timeout 10 bash -c "while ! systemctl is-active --quiet $UNIT_NAME; do sleep 0.1; done"
```

Comment:
use Type=notify and NotifyAccess=all and call systemd-notify --ready before sleeping, instead of looping which is racy and will fail on slow systems